quickstart/101-azure-virtual-desktop-configure/main.tf

```diff
@@ -0,0 +1,63 @@
+# Randomization of resource group name.
+# Resource group name is output when execution plan is applied.
+resource "random_pet" "rg-name" {
+  prefix    = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name      = random_pet.rg-name.id
+  location  = var.resource_group_location
+}
+
+# Create AVD workspace
+resource "azurerm_virtual_desktop_workspace" "workspace" {
+  name                = var.workspace
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = var.deploy_location
+  friendly_name       = "${var.prefix} Workspace"
+  description         = "${var.prefix} Workspace"
+}
+
+# Create AVD host pool
+resource "azurerm_virtual_desktop_host_pool" "hostpool" {
+  resource_group_name      = azurerm_resource_group.rg.name
+  name                     = var.hostpool
+  description              = var.hostpool
+  location                 = var.deploy_location
+  validate_environment     = false #[true false]
+  type                     = "Pooled" #[Pooled Personal]
+  maximum_sessions_allowed = 16 
+  load_balancer_type       = "DepthFirst" #[BreadthFirst DepthFirst]
+  friendly_name            = var.hostpool
+  custom_rdp_properties    = "audiocapturemode:i:1;audiomode:i:0;"
+  preferred_app_group_type = "Desktop" #[Desktop RemoteApp]
+  start_vm_on_connect      = "true"
+ }
+
+# Create registration info
+resource "time_rotating" "avd_token" {
+  rotation_days = 30
+}
+
+resource "azurerm_virtual_desktop_host_pool_registration_info" "registrationinfo" {
+  hostpool_id     = azurerm_virtual_desktop_host_pool.hostpool.id
+  expiration_date = time_rotating.avd_token.rotation_rfc3339
+}
+
+# Create AVD DAG
+resource "azurerm_virtual_desktop_application_group" "dag" {
+  resource_group_name = azurerm_resource_group.rg.name
+  host_pool_id        = azurerm_virtual_desktop_host_pool.hostpool.id
+  location            = var.deploy_location
+  type                = "Desktop"
+  name                = "${var.prefix}-dag"
+  friendly_name       = "Desktop AppGroup"
+  description         = "AVD application group"
+  depends_on          = [azurerm_virtual_desktop_host_pool.hostpool, azurerm_virtual_desktop_workspace.workspace]
```

Comment:
This depends_on should not be required. The host_pool_id property creates an implicit dependency to the host pool and if I am reading the config correctly, the application group and workspace can be created at the same time because the association below connects them afterwards.